docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,306 @@
+resources will be `500m` multiplied by `5`. This of course, can have implications for billing.
+
+Also, with the current `spec.resources` API we have in the `Build`, we are constrain to only define a single set of values for resources (_e.g. memory, cpu, etc_). This is not flexible while
+there could be situations where containers under the same strategy required different resource values (_not sharing the same values_). This raise the following questions:
+
+- First, why do we have in the `Build`/`BuildRun` an API to define `resources` for containers, if the end-users neither know the number of containers for a particular strategy, nor have insights into what each of these containers is doing.
+
+- Second, if a strategy consists of five steps(_containers_) and because of X reason I need to define five different values for each container CPU, how could I achieve this via the current `spec.build` API. (_not flexible_)
+
+## Motivation
+
+For strategies with multiple steps like Buildpacks, not all the steps(_containers_) will require the same set of resources(_e.g. memory, cpu, etc_) values. 
+In order to be able to manage situations where setting particular steps resources is required, we need a more flexible way to deal with N number of steps under the same strategy (_Build/ClusterBuildStrategy_). Also, we still need to help users to know that independently of the abstraction of the strategies, there are options for speeding up the builds. This options could be presented in the form of different `flavours` of the same strategy(_cluster or namespaced scope_), that differ between each other in terms of container resources values.
+
+### Goals
+
+- Keep abstracting the `end-user` from the strategies content.
+
+- Allow `strategy admins` to set different flavours of the **same** strategies. This is the `strategy admin` responsability, and these flavours may vary depending on the Cloud provider. Recommendations
+of different flavours for each strategy could be made, based on the steps and the notion around, which of them consume higher numbers for cpu/memory.
+
+- Set `default` strategies flavours in this repository, so that all strategy containers have limits on the resources they consume, avoiding OOM and CPU throttling when running in Travis.
+
+- Modify the `Build` and the `BuildRun` API, to remove the `spec.resources` configuration path.
+
+- Ensure that the `resources` defined in each `_Build/ClusterBuildStrategy_` will actually be processed and passed on runtime to the generated `TaskRun` from Tekton.
+
+### Non-Goals
+
+- Defining the concrete numbers for the resources(_e.g. CPU/memory_) used per steps in the different Build/ClusterBuildStrategy strategies.
```

Comment:
You mean, it is a non-goal to define the strategy specific resource requirements ? Fine by me, text could look a bit better on this.

Reply:
let me improve on this, danke(german for thanks, lol)

Reply:
this was actually wrong, we do need to modify values, as seen in here:

> 3. For all the current strategies(_BuildStrategy or ClusterBuildStrategy_) in this repository (_samples_), we should define default values for the container resources limits/request so that our `travis` will not be affected by strategy containers consuming all the CPU and memory (_unlimited_). Also, update all of our sample `Build` YAMLs to use this strategies.

because of the above, I´m removing the content inside `Non-Goals` , while is not correct.